translator/translate/logs/logs_collected/files/collect_list/ruleTimestampFormat_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package collect_list
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestApplyTimestampFormatRule(t *testing.T) {
+	r := new(TimestampRegax)
```

Comment:
nit: misspelling
```suggestion
	r := new(TimestampRegex)
```

Reply:
That's already defined here, I was using the existing interface https://github.com/aws/amazon-cloudwatch-agent/blob/70c071bb9093bd4f63a7630abb002f8660ea8341/translator/translate/logs/logs_collected/files/collect_list/ruleTimestampFormat.go#L138

Reply:
Refractored the regex interfacce